dynawo/sources/Models/Modelica/Dynawo/Examples/DynaFlow/IllustrativeExamples/PhaseShifterTransformer.mo

```diff
@@ -0,0 +1,114 @@
+  Dynawo.Electrical.Controls.Basics.SetPoint QrefPu_load_01(Value0 = Q0Pu_load_01);
+
+  // Buses
+  Dynawo.Electrical.Buses.Bus bus_01 annotation(
+    Placement(transformation(origin = {0, 34}, extent = {{-12, -12}, {12, 12}})));
+  Dynawo.Electrical.Buses.Bus bus_02 annotation(
+    Placement(transformation(origin = {0, -44}, extent = {{-12, -12}, {12, 12}})));
+  Dynawo.Electrical.Buses.Bus bus_03 annotation(
+    Placement(transformation(origin = {20, -6}, extent = {{-12, -12}, {12, 12}})));
+
+  // Phase Tap Changer Transformer
+  Dynawo.Electrical.Transformers.TransformerPhaseTapChanger tfo_01(BPu = 0.0*ZBASE69_0, GPu = 0.0*ZBASE69_0, RPu = 0.0/ZBASE69_0, XPu = 0./ZBASE69_0, RatioTfoPu = 1.0, NbTap = 19, Tap0 = 9, ThetaTfoMin = -0.19634954084936207, ThetaTfoMax = 0.19634954084936207) annotation(
+    Placement(transformation(origin = {20, 12}, extent = {{-12, -12}, {12, 12}}, rotation = -90)));
+
+  // Lines
+  Dynawo.Electrical.Lines.Line line_01(BPu = 0.*ZBASE69_0, GPu = 0.*ZBASE69_0, RPu = 0.15/ZBASE69_0, XPu = 18/ZBASE69_0) annotation(
+    Placement(transformation(origin = {-6, -20}, extent = {{-10, -10}, {10, 10}}, rotation = -90)));
+  Dynawo.Electrical.Lines.Line line_02(BPu = 0.*ZBASE69_0, GPu = 0.*ZBASE69_0, RPu = 0.15/ZBASE69_0, XPu = 18/ZBASE69_0) annotation(
+    Placement(transformation(origin = {20, -20}, extent = {{-10, -10}, {10, 10}}, rotation = -90)));
+
+  // Controller
+  Dynawo.Electrical.Controls.Transformers.PhaseShifterP shift_controller_01(P0 = 0.7, PDeadBand = 0.05, PTarget = 0.7, state0 = Dynawo.Electrical.Controls.Transformers.BaseClasses.TapChangerPhaseShifterParams.State.Standard, t1st = 25, tNext = 15, tap0 = 9, tapMax = 18, tapMin = 0, increaseTapToIncreaseValue = false);
+
+  // output connectors
+  Dynawo.Connectors.ImPin PPu_line_01 "Active power transitting in line 01 in pu (base SnRef)";
+
+equation
+  connect(bus_01.terminal, line_01.terminal1) annotation(
+    Line(points = {{0, 34}, {-6, 34}, {-6, -10}}, color = {0, 0, 255}));
+  connect(bus_01.terminal, tfo_01.terminal1) annotation(
+    Line(points = {{0, 34}, {20, 34}, {20, 24}}, color = {0, 0, 255}));
+  connect(line_01.terminal2, bus_02.terminal) annotation(
+    Line(points = {{-6, -30}, {0, -30}, {0, -44}}, color = {0, 0, 255}));
+  connect(line_02.terminal2, bus_02.terminal) annotation(
+    Line(points = {{20, -30}, {2, -30}, {2, -44}, {0, -44}}, color = {0, 0, 255}));
+  connect(gen_01.terminal, bus_01.terminal) annotation(
+    Line(points = {{2, 66}, {0, 66}, {0, 34}}, color = {0, 0, 255}));
+  connect(load_01.terminal, bus_02.terminal) annotation(
+    Line(points = {{0, -70}, {0, -44}}, color = {0, 0, 255}));
+  connect(shift_controller_01.PMonitored, PPu_line_01);
+
+  //PhaseShifter
+  shift_controller_01.locked = if time < 50 then true else false;
+  PPu_line_01.value = -line_01.P2Pu;
+  when shift_controller_01.tap.value <> pre(shift_controller_01.tap.value) then
+    tfo_01.tap.value = shift_controller_01.tap.value;
```

Comment:
tfo_01.tap.value is different than the shift_controller_01.tap.value at the end 
tfo_01.tap.value have the correct value but  shift_controller_01.tap.value doesn't seem to updated

Reply:
Issue opened : #3246